internal/bgp/frr/config.go

```diff
@@ -261,14 +241,34 @@ func templateConfig(data interface{}) (string, error) {
 	i := 0
 	currentCounterName := ""
 	t, err := template.New("FRR Config Template").Funcs(
-		template.FuncMap{"counter": func(counterName string) int {
-			if currentCounterName != counterName {
-				currentCounterName = counterName
-				i = 0
-			}
-			i++
-			return i
-		}}).Parse(configTemplate)
+		template.FuncMap{
+			"counter": func(counterName string) int {
+				if currentCounterName != counterName {
+					currentCounterName = counterName
+					i = 0
+				}
+				i++
+				return i
+			},
+			"ipCmd": func(ipFamily string) string {
+				if ipFamily == "ipv6" {
+					return ipFamily
+				}
+				return "ip"
+			},
+			"prefixesForLocalPref": func(advConfig advertisementConfig, ipFamily string) []localPrefPrefixes {
```

Comment:
This is more "localPrefPrefixesForAdv". Also, can't you use the ipfamily in advertisementConfig?

Reply:
i need the neighbor ip family to avoid mixing ipv4 prefixes with neighbor ipv6 addresses

Reply:
> 

which is something we do not support today right? So that can be solved by filtering when we build the adv list for neighbour?